x/pairing/keeper/staking.go

```diff
@@ -147,7 +146,7 @@ func (k Keeper) StakeNewEntry(ctx sdk.Context, creator, chainID string, amount s
 
 	k.epochStorageKeeper.AppendStakeEntryCurrent(ctx, chainID, stakeEntry)
 
-	err = k.dualstakingKeeper.Delegate(ctx, senderAddr.String(), senderAddr.String(), chainID, amount)
+	err = k.dualstakingKeeper.DelegateFull(ctx, senderAddr.String(), validator, senderAddr.String(), chainID, amount)
```

Comment:
we need to verify TokenDemon and staking token are the same somewhere

Reply:
i will create a task to remove the token denom and start using the bondtoken from the staking module